source/material_model/melt_phipps_morgan.cc

```diff
@@ -0,0 +1,1363 @@
+          // TODO PM01 Only read these in for the solid endmembers
+          prm.declare_entry ("Molar masses", "0.20379, 0.14073, 0.20379, 0.14073",
+                             Patterns::List(Patterns::Double(0)),
+                             "Molar masses of the different endmembers"
+                             "Units: kg/mol.");
+          prm.declare_entry ("Number of atoms", "7.0, 7.0, 7.0, 7.0",
+                             Patterns::List(Patterns::Double(0)),
+                             "Number of atoms in the formula of each endmember."
+                             "Units: none.");
+          // Phipps Morgan Table 1A, olivine, V given in [cm3/mol], converted to [m3/mol]
+          prm.declare_entry ("Reference volumes", "46.39e-6, 43.79e-06, 46.9e-06,  46.9e-06",
+                             Patterns::List(Patterns::Double(0)),
+                             "Reference volumes of the different endmembers."
+                             "Units: $m^3/mol$.");
+          // Holland & Powell Table 2a, assuming a factor 1e-5 Q!
+          prm.declare_entry ("Reference thermal expansivities", "2.82e-5, 2.85e-5, 10.71e-5, 9.2e-5",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of thermal expansivities for each different endmember at the reference temperature "
+                             "and reference pressure."
+                             "Units: 1/K.");
+          // Holland & Powell Table 2a, assuming they are given in [kbar = 1e8 Pa] Q!
+          prm.declare_entry ("Reference bulk moduli", "1.256e11, 1.285e11, 0.29e11, 0.362e11",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of bulk moduli for each different endmember at the reference temperature "
+                             "and reference pressure."
+                             "Units: Pa.");
+          // Holland & Powell Table 2a
+          prm.declare_entry ("First derivatives of the bulk modulus", "4.68, 3.84, 10.42, 10.06",
+                             Patterns::List(Patterns::Double()),
+                             "The pressure derivative of the bulk modulus at the reference temperature and reference "
+                             "pressure for each different endmember component."
+                             "Units: none.");
+          // Holland & Powell Table 2a, assuming they are given in [1/kbar = 1/(1e8 Pa)] Q!
+          prm.declare_entry ("Second derivatives of the bulk modulus", "-3.7e-11, -3.0e-11, -3.59e-10, -2.78e-10",
+                             Patterns::List(Patterns::Double()),
+                             "The second pressure derivative of the bulk modulus at the reference temperature and reference "
+                             "pressure for each different endmember component."
+                             "Units: 1/Pa.");
+          // Q!
```

Comment:
Bottom of page 346 of HP2011:
`Einstein_T = 10636 /  (S0 / n -  6.44)`
S0 is the standard state entropy, and n is the number of atoms in the formula unit (7 for olivine).
This is empirical, so I prefer to provide Einstein_T as an explicit parameter.